backend/common/utils/leaderboard.py

```
@@ -0,0 +1,73 @@
+
+Computes student rankings for a cohort and exports them to CSV.
+Used by the professor dashboard leaderboard widget.
+"""
+import csv
+import sqlite3
+from sqlalchemy.orm import Session
+from common.db.models import User, Simulation
+
+
+# Default grade weighting per simulation type
+def compute_weighted_scores(raw_scores, weights={}):
```

Ruff (0.15.17)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)